### PR TITLE
delay VRControls initialization for node

### PR DESCRIPTION
**Pull request by kellyselden**
In response to https://github.com/aframevr/aframe/pull/2755#discussion_r121292464.

This makes running in node more happy and eliminates the need for this change https://github.com/aframevr/aframe/pull/2755/files#diff-97c6b758b49a87147586d2d87795312aR38.

cc @ngokevin

### CODE REVIEW

File: src/utils/device.js [hunk 1]
```diff
@@ -1,12 +1,21 @@
 var THREE = require('../lib/three');
 var dolly = new THREE.Object3D();
-var controls = new THREE.VRControls(dolly);
+
+var controls;
```
Comment (ngokevin) [inline src/utils/device.js]: Maybe to hide the controls variable:

```js
var getControls = (function () {
  var controls;
  return function getControls () {
    if (!controls) { 
      controls = new THREE.VRControls(dolly);
    }
    return controls;
  };
})();
```

Comment (kellyselden) [inline src/utils/device.js]: good call

Comment (kellyselden) [inline src/utils/device.js]: updated

**Comment by dmarcos** [comment]
This PR has regressed VR mode on desktop. Neither Oculus nor Vive enter VR anymore. Delaying the initialization of VRControls introduce timing issues. Note that VRControls is also asynchronous and we have to somehow handle that asynchrony. The timing was right and the extra delay that this PR introduces affects the initialization timing. This PR is not critical for 0.6.0. I'm going to revert and we revisit later in a follow up PR. @kellyselden feel free to reopen. Make sure that Vive / Oculus enter VR mode.

**Comment by kellyselden** [comment]
Sounds good. I'll take a look.

**Comment by ngokevin** [comment]
Perhaps try checking if it's a node environment and not initializing it if so?

**Comment by kellyselden** [comment]
Also, if you can think of a test that would have caught this regression, I would be happy to include/write it.

**Pull request merged**